core/station/impl/src/models/request_policy.rs

```diff
@@ -150,4 +222,95 @@ pub mod request_policy_test_utils {
             crate::models::RequestStatus::Approved
         ));
     }
+
+    #[test]
+    fn test_invalid_rule_for_specifier() {
+        let incompatible_rule_1 = NAMED_RULE_SERVICE
+            .create(AddNamedRuleOperationInput {
+                name: "test_1".to_string(),
+                description: None,
+                rule: RequestPolicyRule::And(vec![
+                    RequestPolicyRule::AllowListed,
+                    RequestPolicyRule::AutoApproved,
+                ]),
+            })
+            .expect("Failed to create named rule");
+
+        let incompatible_rule_2 = NAMED_RULE_SERVICE
+            .create(AddNamedRuleOperationInput {
+                name: "test_2".to_string(),
+                description: None,
+                rule: RequestPolicyRule::AllowListedByMetadata(MetadataItem {
+                    key: "test".to_string(),
+                    value: "test".to_string(),
+                }),
+            })
+            .expect("Failed to create named rule");
+
+        let tests = [
```

Comment:
How about only listing the rules and errors here and later do a "product" test:
```
for (specifier, expect_err) in [(RequestSpecifier::AddAccount, true), (RequestSpecifier::Transfer, false)] {
  for (rule, err) in tests {
    // ...
    let result = policy.validate();
    if expect_err {
      assert_eq!(result, expected);
    } else {
      result.expect("...");
    }
  }
}
```

Reply:
I wanted to increase coverage (without a lot of copying), but I see your point and don't have a strong opinion here.